provider/resource_generic.go

```diff
@@ -39,27 +39,21 @@ func genericResourceRead(rKind model.Kind, d *schema.ResourceData, meta any) err
 	// A nil return from GenericResource indicates that the resource
 	// did not exist. Terraform read operations should always set the
 	// ID to "" and return a nil error. This will allow Terraform to
-	// re-create the resource or comfirm that it was deleted.
+	// re-create the resource or confirm that it was deleted.
 	if g == nil {
 		d.SetId("")
 		return nil
 	}
 
-	// Save values returned by bindplane to Terraform's state
-
-	d.SetId(g.ID)
```

Comment:
We do not need to set the resource ID in the state anymore. Existing resources will already have an ID in the state and new resources will have had their ID generated during create.